plugin/pkg/shoot/validator/admission.go

```diff
@@ -1229,6 +1230,10 @@ func validateZones(constraints []core.Region, region, oldRegion string, worker,
 				allErrs = append(allErrs, field.NotSupported(jdxPath, zone, validZones))
 			}
 		}
+		if usedZones.Has(zone) {
+			allErrs = append(allErrs, field.Duplicate(jdxPath, zone))
```

Comment:
Since this is an incompatible change, I'd vote for having this check only for new clusters or new worker pools because this issue cannot be easily fixed for existing clusters, concretely zone removal is not supported.

Reply:
The zones are only validated if the worker or the region is changing, see line https://github.com/gardener/gardener/blob/master/plugin/pkg/shoot/validator/admission.go#L1219

Isn't that good enough?

Reply:
We looked briefly at the code and in contrast to https://github.com/gardener/gardener-extension-provider-aws/issues/643 here the validation seems like it is actually only called when worker or region changed.

Reply:
Thanks for pointing this out, seems to be fine then.